### PR TITLE
intrinsics base data concept for node

### PR DESCRIPTION
**Pull request by viperscape**

### CODE REVIEW

File: src/node.rs [hunk 1]
```diff
@@ -1,215 +1,38 @@
...
-        self
-    }
-
-    fn label_font_size(mut self, size: FontSize) -> Self {
-        self.style.maybe_label_font_size = Some(size);
-        self
-    }
+#[derive(Debug)]
+pub struct Edge;
+impl EditableEdge for Edge {
```
Comment (stjahns) [inline src/node.rs:15]: My original intent was for this to not be implemented by the library itself, but rather users of the library, as different users might need different data associated with an edge, e.g. a 'weight' or a 'transition condition' etc. But a simple edge with no data might be nice to have in here anyway.

File: src/nodewidget.rs [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+        }
+    }
+}
+
+impl<'a> CustomWidget for Node<'a> {
+
+    type State = NodeState;
+
+    fn update<C>(self, state: NodeState, ui_id: UiId, ui: &mut Ui<C, Node<'a>>) -> WidgetUpdate<NodeState>
+        //where C: CharacterCache //I had to comment this out, not sure why it wouldn't compile for me
```
Comment (stjahns) [inline src/nodewidget.rs:97]: I sent a PR to Conrod, I think it was merged today, so if you update it should work again.

**Comment by stjahns** [comment]
After thinking about this, I think we are on a similar track, except I don't want to have to force users to store editor session data as a member in their own data, which seems to be the intent of 

```
// custom node struct
struct Node {
    some_state: bool,
    base: NodeState, //must implement some base state, and impl it
}
```

Instead, I want to try out the idea of building a copy of the user's graph (using petgraph), where each node in our internal graph wraps the data from the original graph, while adding additional editor data like the corresponding conrod widget ID. The user graph nodes will still need to implement something like EditableNode for getting/setting position, as we will need to be able to serialize/deserialize positions in order for graph layouts to be persistent, but otherwise they shouldn't have to alter their data structures in order to be used with this editor (ideally). Ill give it a quick shot to see how it goes..

**Comment by stjahns** [comment]
whoops, i was going to merge the node -> nodewidget rename. I think i'll just add a `widgets` module and stick it in there.

**Comment by viperscape** [comment]
> I don't want to have to force users to store editor session data as a member in their own data

That makes sense

**Pull request closed without merging**